deepchecks/nlp/utils/text_data_plot.py

```diff
@@ -0,0 +1,201 @@
+                            text='<b>10<sup>th</sup> Percentile</b>', showarrow=False, font={'size': 12}))
+
+    shapes.append(dict(type='line', x0=percentile_90, y0=0, x1=percentile_90, y1=max(y_value),
+                       line={'color': feature_distribution_colors['measure'], 'dash': 'dashdot', 'width': 3}))
+    annotations.append(dict(x=percentile_90 + max(x_value)*0.02, y=max(y_value)/2, textangle=-90,
+                            text='<b>90<sup>th</sup> Percentile</b>', showarrow=False, font={'size': 12}))
+
+    xaxis_layout = dict(title=property_name)
+    yaxis_layout = dict(title='Density')
+
+    return trace, shapes, annotations, xaxis_layout, yaxis_layout
+
+
+def text_data_describe_plot(all_properties_data, n_samples, label, categorical_metadata, numerical_metadata,
+                            categorical_properties, numerical_properties, properties: List[str]):
+    """Return a plotly figure instance.
+
+    Parameters
+    ----------
+    all_properties_data:
+    n_samples:
+    label:
+    categorical_metadata:
+    numerical_metadata:
+    categorical_properties:
+    numerical_properties:
+    properties : List[str]
+        List of property names to consider for generating property distribution graphs.
+
+    Returns
+    -------
+    Plotly Figure instance.
+    """
+    specs = [[{'type': 'pie'}, {'type': 'table'}]] + \
+        [[{'type': 'xy', 'colspan': 2}, None] for _ in range(len(properties))]
+
+    subplot_titles = [f'Label Distribution<br><sup>Out of {pd.notna(label).sum()} annotated samples</sup><br><br>',
+                      '']
+    for prop in properties:
+        if prop in TEXT_PROPERTIES_DESCRIPTION:
+            subplot_titles.append(f'{prop} Property Distribution<sup><a href="{get_docs_link()}nlp/usage_guides/'
+                                  'nlp_properties.html#deepchecks-built-in-properties">&#x24D8;</a></sup><br>'
+                                  f'<sup>{TEXT_PROPERTIES_DESCRIPTION[prop]}</sup>')
+
+    fig = make_subplots(rows=len(properties) + 1, cols=2, specs=specs, subplot_titles=subplot_titles,
+                        row_heights=[1.5] + [1.0] * len(properties))
+    label_counts = pd.Series(label).value_counts()
+
+    # Pie chart for label distribution
+    fig.add_trace(go.Pie(labels=list(label_counts.index), values=list(label_counts), textposition='inside',
+                         hovertemplate='%{label}: %{value} samples<extra></extra>', textinfo='label+percent',
+                         showlegend=False), row=1, col=1)
+
+    # Table figure for displaying some statistics
+    table_trace = _generate_table_trace(n_samples, label, categorical_metadata, numerical_metadata,
+                                        categorical_properties, numerical_properties)
+    fig.add_trace(table_trace, row=1, col=2)
+
+    # Looping over all the properties to generate respective property distribution graphs
```

Comment:
I'm missing a title between the label distribution and the properties one. Something that says "these are the properties distributions", and also maybe links to the property guide.

Also, what happens in the case there are no properties, but there's metadata and embeddings? We would still show nothing, and that's a shame... We can easily show metadata for instance

Reply:
If no properties are there, we will still show label distribution and the table next to the label distribution.
Just to make sure, we are only displaying property distribution and not metadata distribution. Do we want this as well?